sklearn/feature_selection/mutual_info.py

```diff
@@ -0,0 +1,375 @@
+        within (0.0, 1.0) is interpreted as the percentage from the initial
+        number of features (rounded down). Half of the features is selected by
+        default. After the class instance was fitted you can decrease
+        `n_features_to_select` and correctly select a smaller subset of
+        features without refitting.
+
+    use_redundancy : bool, default True
+        If True, then "min-redundancy max-relevance" algorithm will be used,
+        otherwise features will be selected only based on relevance, i.e.
+        mutual information between features and the target variable.
+
+    categorical_features : bool or array_like with shape (n_features),
+                           default False
+        If bool, then determines whether to consider all features categorical
+        or continuous. If array, then True elements mark categorical features.
+
+    categorical_target : bool, default False
+        Whether to consider `y` as a categorical variable.
+
+    n_neighbors : int, default 3
+        Number of neighbors to use for mutual information estimation for
+        continuous variables, see [2]_ and [3]_. The default value is usually
+        sufficient.
+
+    copy : bool, default True
+        Whether to make a copy of the given data. If set to False, the initial
+        data will be overwritten if it was in dense format.
+
+    Attributes
+    ----------
+    ranks_ : ndarray, shape (n_features,)
+        Ranks of features, values from 0 (best) to n_features-1 (worst).
+        If ``use_redundancy=True``, then only `n_features_to_select` values
+        were actually filled, the rest are filled with n_features - 1.
+
+    References
+    ----------
+    .. [1] H. Peng, F. Long, and C. Ding, "Feature selection based on mutual
+           information: criteria of max-dependency, max-relevance, and
+           min-redundancy", IEEE Transactions on Pattern Analysis and Machine
+           Intelligence, Vol. 27, No. 8, pp. 1226-1238, 2005.
+    .. [2] A. Kraskov, H. Stogbauer and P. Grassberger, "Estimating mutual
+           information". Phys. Rev. E 69, 2004.
+    .. [3] B. C. Ross "Mutual Information between Discrete and Continuous
+           Data Sets". PLoS ONE 9(2), 2014.
+    """
+    def __init__(self, n_features_to_select=0.5, use_redundancy=True,
+                 categorical_features=False, categorical_target=False,
+                 n_neighbors=3, copy=True):
+        self.n_features_to_select = n_features_to_select
+        self.use_redundancy = use_redundancy
+        self.categorical_features = categorical_features
+        self.categorical_target = categorical_target
+        self.n_neighbors = n_neighbors
+        self.copy = copy
+
+    def _get_support_mask(self):
+        n_features_to_select = self._get_n_features_to_select()
+        support = np.zeros_like(self.ranks_, dtype=bool)
+        support[self.ranks_ < n_features_to_select] = True
+        return support
```

Comment:
The problem is that `n_features_to_select` might be float between 0 and 1, so I need to recompute it in this strange way.

About your code sample: `support` should be a boolean mask with size `X.shape[1]`, right?